mantis-control-plane/mantis-control-plane-server/src/main/java/io/mantisrx/server/master/persistence/IMantisPersistenceProvider.java

```diff
@@ -165,6 +165,12 @@ void storeAndUpdateWorkers(final IMantisWorkerMetadata existingWorker, final IMa
 
     List<DisableTaskExecutorsRequest> loadAllDisableTaskExecutorsRequests(ClusterID clusterID) throws IOException;
 
+    void storeNewDisabledTaskExecutor(TaskExecutorID taskExecutorID) throws IOException;
+
+    void deleteDisabledTaskExecutor(TaskExecutorID taskExecutorID) throws IOException;
+
+    List<TaskExecutorID> loadAllDisabledTaskExecutors(ClusterID clusterID) throws IOException;
+
```

Comment:
At the storage layer, we can use the existing methods to keep this simple and reduce the surface area.